.github/workflows/integration_tests.yml

```diff
@@ -58,18 +60,34 @@ jobs:
           virtualenvs-create: true
           virtualenvs-in-project: true
           installer-parallel: true
+      - name: Cache Python virtualenv
+        id: cache-poetry-venv
+        uses: actions/cache@v4
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: |
+            venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-
       # - name: Load cached venv
       #   id: cached-poetry-dependencies
       #   uses: actions/cache@v4
       #   with:
       #     path: .venv
       #     key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
       - name: Install dependencies
-        # if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        if: steps.cache-poetry-venv.outputs.cache-hit != 'true'
         run: poetry install --no-interaction --no-root
+      - name: Cache Playwright browsers
+        id: cache-playwright
+        uses: actions/cache@v4
+        with:
+          path: .playwright
```

Comment:
Probably we should use the `PLAYWRIGHT_BROWSERS_PATH` env variable since we have it